azure/scope/managedcontrolplane.go

```diff
@@ -19,14 +19,18 @@ package scope
 import (
 	"context"
 	"encoding/json"
+	"fmt"
 	"strings"
 	"time"
 
 	asoresourcesv1 "github.com/Azure/azure-service-operator/v2/api/resources/v1api20200601"
 	"github.com/pkg/errors"
 	"golang.org/x/mod/semver"
+	"gopkg.in/yaml.v2"
```

Comment:
Could we import v3 here instead since it looks like we already use that through some dependency elsewhere? That seems to have a higher likelihood of getting updates down the line.